### ISSUE TITLE
Kontena 1.2 fails to re-schedule stateful ha service instance on removed node

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#2274]
With a `--stateful` service using the default `--deploy-strategy ha`, with an instance deployed to a host node:

```
$ kontena service create --deploy-strategy ha --stateful redis-ha-stateful redis
 [done] Creating redis-ha-stateful service      
$ kontena service deploy redis-ha-stateful
 [done] Deploying service redis-ha-stateful      
⊛ Deployed instance development/null/redis-ha-stateful-1 to node core-02
```

After disconnecting and removing that host node, the service fails to deploy:

```
$ kontena service deploy redis-ha-stateful
 [fail] Deploying service redis-ha-stateful      
 [error] Kontena::Errors::StandardErrorArray : Cannot find applicable node for service instance development/null/redis-ha-stateful-1: Strategy Scheduler::Strategy::HighAvailability did not find any node:
```

Expected behaviour re Kontena 1.1 would be for the stateful service instance to get rescheduled to a different host node. After the host node is removed, Kontena 1.1 [would also destroy the containers](https://github.com/kontena/kontena/blob/v1.1.5/server/app/models/host_node.rb#L45-L47) for that host node, including the stateful volume containers used by the schduler.

In Kontena 1.2, destroying the `HostNode` only [marks the `GridServiceInstance.host_node` as `nil`](https://github.com/kontena/kontena/blob/v1.2/server/app/models/host_node.rb#L36), which means that the [`Scheduler::Strategy::HighAvailablility#find_stateful_node`](https://github.com/kontena/kontena/blob/v1.2.1/server/app/services/scheduler/strategy/high_availability.rb#L47) will still pick up that `GridServiceInstance`,  and find a `nil` host node:

```ruby
        prev_instance = grid_service.grid_service_instances.find_by(
          instance_number: instance_number
        )
        if prev_instance
          nodes.find{ |n| n.node == prev_instance.host_node }
        ...
```

**Comment by SpComb** [comment]
The new `kontena service remove --instance` API does not serve as a workaround, because it also just sets `GridServiceInstance.host_node_id = nil`, leaving the service in exactly the same state, and causing stateful scheduling to fail.

**Comment by SpComb** [comment]
Tested as fixed in 1.2.2.dev by #2256 

```
$ kontena service show redis-ha-stateful
development/null/redis-ha-stateful:
  created: 2017-05-08T10:55:55.980Z
  updated: 2017-05-08T10:55:55.980Z
  stack: development/null
  desired_state: running
  image: redis:latest
  revision: 1
  stateful: yes
  scaling: 1
  strategy: ha
  deploy_opts:
    min_health: 0.8
  dns: redis-ha-stateful.development.kontena.local
  instances:
    redis-ha-stateful/1:
      scheduled_to: core-02
      deploy_rev: 2017-05-08 10:56:06 UTC
      rev: 2017-05-08 10:56:06 UTC
      state: running
      containers:
$ kontena node rm --force core-02
 [done] Removing core-02 node from development grid      
$ kontena service deploy redis-ha-stateful
 [done] Deploying service redis-ha-stateful      
⊛ Deployed instance development/null/redis-ha-stateful-1 to node core-01
```
